src/pybind/mgr/rest/app/views/v2.py

```diff
@@ -0,0 +1,770 @@
+the monitor *name* as opposed to the monitor *rank*.
+
+The quorum status reported here is based on the last mon status reported by
+the Ceph cluster, and also the status of each mon daemon queried by Calamari.
+
+For debugging mons which are failing to join the cluster, it may be
+useful to show users data from the /status sub-url, which returns the
+"mon_status" output from the daemon.
+
+    """
+    serializer_class = MonSerializer
+
+    def _get_mons(self):
+        monmap_mons = self.client.get_sync_object(MonMap).data['mons']
+        mon_status = self.client.get_sync_object(MonStatus).data
+
+        for mon in monmap_mons:
+            mon['in_quorum'] = mon['rank'] in mon_status['quorum']
+            mon['server'] = self.client.get_metadata("mon", mon['name'])['hostname']
+            mon['leader'] = mon['rank'] == mon_status['quorum'][0]
+
+        return monmap_mons
+
+    def retrieve(self, request, mon_id):
+        mons = self._get_mons()
+        try:
+            mon = [m for m in mons if m['name'] == mon_id][0]
+        except IndexError:
+            raise Http404("Mon '%s' not found" % mon_id)
+
+        return Response(self.serializer_class(DataObject(mon)).data)
+
+    def list(self, request):
+        mons = self._get_mons()
+        return Response(
+            self.serializer_class([DataObject(m) for m in mons],
+                                  many=True).data)
+
+
+if False:
+    class CliViewSet(RemoteViewSet):
+        """
+    Access the `ceph` CLI tool remotely.
+
+    To achieve the same result as running "ceph osd dump" at a shell, an
+    API consumer may POST an object in either of the following formats:
+
+    ::
+
+        {'command': ['osd', 'dump']}
+
+        {'command': 'osd dump'}
+
+
+    The response will be a 200 status code if the command executed, regardless
+    of whether it was successful, to check the result of the command itself
+    read the ``status`` attribute of the returned data.
+
+    The command will be executed on the first available mon server, retrying
+    on subsequent mon servers if no response is received.  Due to this retry
+    behaviour, it is possible for the command to be run more than once in
+    rare cases; since most ceph commands are idempotent this is usually
+    not a problem.
+        """
+        serializer_class = CliSerializer
+
+        def create(self, request):
+            # Validate
+            try:
+                command = request.DATA['command']
+            except KeyError:
+                raise ParseError("'command' field is required")
+            else:
+                if not (isinstance(command, basestring) or isinstance(command, list)):
+                    raise ParseError("'command' must be a string or list")
+
+            # Parse string commands to list
+            if isinstance(command, basestring):
+                command = shlex.split(command)
+
+            name = self.client.get_cluster(fsid)['name']
```

Comment:
I've added a commit to strip out the `if False` blocks and/or replace them with django-rest-framework version checks as appropriate